editor/code_editor.cpp

```diff
@@ -1608,6 +1618,94 @@ void CodeTextEditor::duplicate_selection() {
 	text_editor->queue_redraw();
 }
 
+void CodeTextEditor::duplicate_lines(bool p_below) {
+	text_editor->begin_complex_operation();
+
+	Vector<int> caret_edit_order = text_editor->get_caret_index_edit_order();
+	for (const int &caret_index : caret_edit_order) {
+		// The text that will be inserted. All lines in one string
+		String insert_text;
+
+		// The new line position of the caret after the operation
+		int new_caret_line = text_editor->get_caret_line(caret_index);
+		// The new column position of the caret after the operation
+		int new_caret_column = text_editor->get_caret_column(caret_index);
+		// The caret positions of the selection. Stays -1 if there is no selection
+		int select_from_line = -1;
+		int select_to_line = -1;
```

Comment:
You don't use these much outside of places where `has_selection()` is checked. You should define them close to the scope where the variables have a meaning.

Reply:
If I put them in their own scope I would need to write duplicate code.